src/app/wallets/update-pending-payments.ts

```diff
@@ -0,0 +1,188 @@
+import { PaymentStatus } from "@domain/bitcoin/lightning"
+import { InconsistentDataError } from "@domain/errors"
+import { toLiabilitiesAccountId } from "@domain/ledger"
+import { FeeReimbursement } from "@domain/ledger/fee-reimbursement"
+import { LedgerService } from "@services/ledger"
+import { LndService } from "@services/lnd"
+import { LockService } from "@services/lock"
+import { PriceService } from "@services/price"
+
+export const updatePendingPayments = async ({
+  walletId,
+  logger,
+  lock,
+}: {
+  walletId: WalletId
+  logger: Logger
+  lock?: DistributedLock
+}): Promise<void | ApplicationError> => {
+  const liabilitiesAccountId = toLiabilitiesAccountId(walletId)
+
+  const lockService = LockService()
+  await lockService.lockWalletId({ walletId, logger, lock }, async () => {
```

Comment:
@bodymindarts I had to leave in the `walletId` argument to pass it forward to the `redlock`